From 7f0e5b3d27053406dcba9779af5604b978e6bf3a Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Sat, 1 Apr 2017 09:08:14 +0800
Subject: [PATCH] arm: dts: rk3288-android: add psci v1.0 support

Add psci v1.0 support to rk3288, support it to call into
secure world through psci APIs.

Change-Id: I1dc98fbdcc544a2bf0ec09544e946756757d2d80
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 arch/arm/boot/dts/rk3288-android.dtsi | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288-android.dtsi b/arch/arm/boot/dts/rk3288-android.dtsi
index 84bd8f3879e8..60363287cbc1 100644
--- a/arch/arm/boot/dts/rk3288-android.dtsi
+++ b/arch/arm/boot/dts/rk3288-android.dtsi
@@ -47,6 +47,11 @@
 		bootargs = "earlycon=uart8250,mmio32,0xff690000";
 	};
 
+	psci {
+		compatible = "arm,psci-1.0";
+		method = "smc";
+	};
+
 	backlight: backlight {
 		compatible = "pwm-backlight";
 		brightness-levels = <
@@ -184,6 +189,22 @@
 	};
 };
 
+&cpu0 {
+	enable-method = "psci";
+};
+
+&cpu1 {
+	enable-method = "psci";
+};
+
+&cpu2 {
+	enable-method = "psci";
+};
+
+&cpu3 {
+	enable-method = "psci";
+};
+
 &dmac_bus_s {
 	/* change to non-secure dmac */
 	reg = <0xff600000 0x4000>;
-- 
2.35.3

